clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,311 @@
+    )
+
+    df_subject_small = (
+        dict_df["subject"]
+        .merge(df_source["Subject"], how="inner", on="Subject")
+        .drop_duplicates()
+    )
+    df_source = df_source.merge(
+        dict_df["mri"]["Scanner"], how="left", left_on="path", right_on="MR ID"
+    )
+
+    df_small = df_subject_small.merge(df_adrc_small, how="inner", on="Subject")
+    df_small = df_small.assign(participant_id=lambda df: "sub-" + df.Subject)
+    df_source = df_source.assign(
+        session=lambda df: (round(df["Date"].str[1:].astype("int") / (365.25 / 2)) * 6)
+    )
+
+    df_source = df_source.assign(
+        ses=lambda df: df.session.apply(
+            lambda x: f"ses-M{ (5-(len(str(x)))) * '0' + str(int(x))}"
+        )
+    )
+
+    df_source = df_source.join(
+        df_source.modality.map(
+            {
+                "MR": {"datatype": "anat", "suffix": "T1w"},
+                "FDG": {"datatype": "pet", "suffix": "pet", "trc_label": "18FFDG"},
+                "PIB": {"datatype": "pet", "suffix": "pet", "trc_label": "11CPIB"},
+                "AV45": {"datatype": "pet", "suffix": "pet", "trc_label": "18FAV45"},
+            }
+        ).apply(pd.Series)
+    )
+    if "trc_label" in df_source.columns:
+        df_source = df_source.assign(
+            filename=lambda df: df.apply(
+                lambda x: f"{x.participant_id}/{x.ses}/{x.datatype}/"
+                f"{x.participant_id}_{x.ses}"
+                f"{'_trc-'+x.trc_label if pd.notna(x.trc_label) else ''}"
+                f"_{x.suffix}.nii.gz",
+                axis=1,
+            )
+        )
+    else:
+        df_source = df_source.assign(
+            filename=lambda df: df.apply(
+                lambda x: f"{x.participant_id}/{x.ses}/{x.datatype}/"
+                f"{x.participant_id}_{x.ses}"
+                f"_{x.suffix}.nii.gz",
+                axis=1,
+            )
+        )
+    df_adrc = df_adrc.merge(df_source["Subject"], how="inner", on="Subject")
+    df_adrc = df_adrc.assign(
+        session=lambda df: round(df["session_id"].str[1:].astype("int") / (364.25 / 2))
+        * 6
+    )
+    df_adrc = df_adrc.drop_duplicates().set_index(["Subject", "session_id"])
+    df_source = df_source.merge(
+        df_adrc[
+            [
+                "session",
+                "mmse",
+                "cdr",
+                "commun",
+                "dx1",
+                "homehobb",
+                "judgment",
+                "memory",
+                "orient",
+                "perscare",
+                "sumbox",
+                "apoe",
+            ]
+        ],
+        how="left",
+        on="session",
+    )
+    return df_source, df_small
+
+
+def dataset_to_bids(
+    df_source: DataFrame, df_small: DataFrame
+) -> Tuple[DataFrame, DataFrame, DataFrame]:
+    # Build participants dataframe
+    df_participants = (
+        df_small[["participant_id", "ageAtEntry", "M/F", "Hand", "Education"]]
```

Comment:
More of an open question but shouldn't we rather have this mapping of name in a separate file (tsv?) for easier manipulation?